components/pedigree/resources/src/main/resources/pedigree/filesaver/FileSaver.js

```diff
@@ -13,264 +13,275 @@
-                              fs_error();
-                          }
-                      }));
-                  }), fs_error);
-              }), fs_error);
           }
-          , FS_proto = FileSaver.prototype
-          , saveAs = function (blob, name) {
-              return new FileSaver(blob, name);
-          }
-      ;
-      FS_proto.abort = function () {
-          var filesaver = this;
-          filesaver.readyState = filesaver.DONE;
-          dispatch(filesaver, "abort");
-      };
-      FS_proto.readyState = FS_proto.INIT = 0;
-      FS_proto.WRITING = 1;
-      FS_proto.DONE = 2;
-
-      FS_proto.error =
-      FS_proto.onwritestart =
-      FS_proto.onprogress =
-      FS_proto.onwrite =
-      FS_proto.onabort =
-      FS_proto.onerror =
-      FS_proto.onwriteend =
-          null;
 
-      view.addEventListener("unload", process_deletion_queue, false);
-      saveAs.unload = function () {
-          process_deletion_queue();
-          view.removeEventListener("unload", process_deletion_queue, false);
-      };
-      return saveAs;
-  }(
-       typeof self !== "undefined" && self
-    || typeof window !== "undefined" && window
-    || this.content
-));
-// `self` is undefined in Firefox for Android content script context
-// while `this` is nsIContentFrameMessageManager
-// with an attribute `content` that corresponds to the window
+          var doc = saveTxtWindow.document;
+          doc.charset = charset;
+          if (browser.versionMajor <= 8) {
+            doc.open('text/plain', 'replace');
+          } else {
+            doc.open();
+          }
+          doc.write(textContent);
+          doc.close();
+          fileName += '.txt';
+          var retValue = doc.execCommand('SaveAs', null, fileName);
+          saveTxtWindow.close();
+          return retValue;
+      }
+  })
 
-var saveTextAs = saveTextAs
-|| (function (textContent, fileName, charset) {
-    fileName = fileName || 'pedigree.txt';
-    charset = charset || 'utf-8';
-    textContent = (textContent || '').replace(/\r?\n/g, "\r\n");
-    if (saveAs && Blob) {
-        var blob = new Blob([textContent], { type: "text/plain;charset=" + charset });
-        saveAs(blob, fileName);
-        return true;
-    } else {//IE9-
-        var saveTxtWindow = window.frames.saveTxtWindow;
-        if (!saveTxtWindow) {
-            saveTxtWindow = document.createElement('iframe');
-            saveTxtWindow.id = 'saveTxtWindow';
-            saveTxtWindow.style.display = 'none';
-            document.body.insertBefore(saveTxtWindow, null);
-            saveTxtWindow = window.frames.saveTxtWindow;
-            if (!saveTxtWindow) {
-                saveTxtWindow = window.open('', '_temp', 'width=100,height=100');
-                if (!saveTxtWindow) {
-                    window.alert('Sorry, download file could not be created.');
-                    return false;
-                }
-            }
-        }
+  var FileSaver = {};
```

Comment:
This is a new object that is exported at the end of this module. Originally this file just defined two global functions, now it exports those functions in this object.